public/app/routes/ReactContainer.tsx

```diff
@@ -16,16 +17,31 @@ function WrapInProvider(store, Component, props) {
 }
 
 /** @ngInject */
-export function reactContainer($route, $location, backendSrv: BackendSrv, datasourceSrv: DatasourceSrv) {
+export function reactContainer(
+  $route,
+  $location,
+  backendSrv: BackendSrv,
+  datasourceSrv: DatasourceSrv,
+  contextSrv: ContextSrv
+) {
   return {
     restrict: 'E',
     template: '',
     link(scope, elem) {
-      let component = $route.current.locals.component;
+      // Check permissions for this component
```

Comment:
You can/should add explore as a backend route as well like here: https://github.com/grafana/grafana/blob/master/pkg/api/api.go#L52 but using *reqEditorRole* instead of *reqGrafanaAdmin*.

Please shout if you need help with the GO changes.